### ISSUE TITLE
Waveshare Pico S3 Config

### ISSUE THREAD

**Issue by MisterDiz** [wezhunter/ESP32_LinuxCNC_MotionController_RealTime#27]
Firstly, I'd like to offer a huuuuuuge thank you for what you are doing. Having the power and affordability (and availability!) of ESP32 chips with LinuxCNC has long been a wish of mine and it is great to see someone attack it with the lustre, dedication and skill that you are bringing.

With your recent missive about using the Waveshare ESP32 S3 Pico, I dived in head long and ordered one and a Pico breakout board. Unfortunately I am getting a little unstuck as to how to get things going. The ESP32 release uploaded but didn't work (unsurprisingly) so I have been tinkering with the platform.ini file to make something work for the ESP32S3. Alas, I'm having no joy. I have tried slashing away at the S2 Pico code but I can't even get that to compile by uncommenting it and other similarly unsophisticated guesswork.

I am also awaiting a more basic WT32-ETH01 board to arrive but would like to use the more capable S3 unit if I can, as I had assumed it was the way forward. I would be grateful if you could point me in the right direction of what to do to make it work, or offer something more up-to-date config-wise, as your videos are too salivating for me to ignore.

TIA!

**Comment by wezhunter** [comment]
Hi,
Thanks for the kind words and support! 

I intend on picking this up again in the very near future - progress stopped a few weeks ago but should restart very soon.

The next release, once I've gotten round to fully testing it, will include S3 support. The current branch that is on GitHub doesn't support either the S2/S3 MCUs.

The reason for the detailed offline testing, on a real machine that I have here, is simply the fact it will generate more questions if not done to the highest of standards and the feature-set can be quite complex at times.

Following on after the release there is a high likelihood that I'll be leaving it up to the community to maintain the project.

Designing an open source EMI compliant/resistant PCB that has level shifting, isolation and IO expansion supporting all the ESP32/S-x series (Ethernet and USB) will is expected to take approximately 1 week of dev effort (part time) which I fully intend on releaseing.

Reasonably priced kits will be made available or you can roll your own.

I'll gladly drop a message on here once I pick it back up again.

Thanks,
Wez

**Comment by MisterDiz** [comment]
Nice one Wez, I look forward eagerly to the next branch release and totally understand that you want to assure its functionality before the internet gets to rip it apart.

I hope that now the baby is born that you won't stray too far but I can understand that its demands could become quite onerous and that there's this other thing that can get in the way, called life. I only regret that I don't have the skills to help push it along. 

Put me down for a PCB/kit, it would be rude not to!

All the best
Diz

**Comment by wezhunter** [comment]
Thanks Diz. The project has come too far to not have it functional. Be nice to eventually see torture test results from others. Hopefully my code comments/docs will help support any future maintainers.

Porting to the upcoming ESP32-P4 MCU should be relatively straightforward so ensures long term sustainability. I am aware of Mesa and NVEM controllers being challenging to source in some geographies.

Lastly, whilst I'm here, I will eventually release a board design that leverages this project but multiple ESP32 ICs to offer 1mhz stepper pulse generation/tracking per-axis & full-quad encoder support (one IC per axis). 
Each IC is synchronised to a master IC which then connects to the LCNC host. 
Mainly for the larger/commercial retrofits or very high microstepping and speeds. 
Partly a bit of a compromise for FPGAs/Mesa controllers - ESP32/Sx/P4 ICs are abundant excellent IO & stability.

Best,
Wez

**Comment by MisterDiz** [comment]
Ahhh, so the P4 is why you were doing LVGL tests?! That with per axis MCUs and quadrature encoders will be kick ass. Fabulous and dream like stuff. 

Mesa's are about as abundant as rocking horse crap around here. Can't wait.

**Comment by wezhunter** [comment]
Yeah in part - the Sx series cannot drive both a high res LVGL/LCD and be a motion controller simultaneously due to IRAM limitations but the P4 theoretically could. Mainly an experiment. Be nice to have a wireless high res wireless touch control panel that is in sync via the motion controller and LCNC host. 

Take a look at #18 for the general concept...

Mesa is great but FPGA supply can be problematic. Also something based on Arduino/Espressif frameworks opens up development & integration possibilities to most too.

Peace out for now...

**Comment by MisterDiz** [comment]
Makes sense. I'm a big fan of SquareLine and those yellow board ESP32 screens and have ESPNow working for a wireless probe. Seems like we're on a similar sheet of hymns... I'll watch out for any updates. Peace!

**Comment by wezhunter** [comment]
Great to hear about the probe nice work!

Drats... my partly secret plan was foiled :) SquareLine is very very good - shall get it working as an LCNC client with the S3/P4 eventually. 
Yellow boards lcds are cool but not a fair comparison to a RGB8080 capacitive glass touch. Pixel density and clarity is considerably better. For info I've tested every possible type of display that and a 800x480 RGB8080 panel is the best experience for the S2/S3.

MIPI DSI on the P4 is going to be the killer peripheral though - embedded 1080P cap touch with LVGL, MP and SquareLine with a S2 slave IC acting as a ESPNow adapter to motion controller. A true high fidelity customisable LCNC embedded HMI with an industrious level of reliability would be lovely.

https://www.youtube.com/watch?v=u5fmZBoeJOU

**Comment by MisterDiz** [comment]
Bring it on!
